### PR TITLE
Update bindings for 0.24

### PR DESCRIPTION
**Pull request by amaanq**
This PR has a few changes stemming from upstream, namely:
 - migrate to `tree-sitter.json`
 - tweak language bindings
 - use `github` workflow for uploading artifacts on release

### CODE REVIEW

**Comment by eed3si9n** [comment]
Could you try removing any generated files from the PR and let the GitHub Action generate it plz? I think it's checking for the fidelity of all files because it detected a new .c file.

**Comment by amaanq** [comment]
> Could you try removing any generated files from the PR and let the GitHub Action generate it plz? I think it's checking for the fidelity of all files because it detected a new .c file.

I didn't add a c file though, let alone any generated ones. I question the fidelity script here (and this is the only grammar repo I know of that uses this)

File: bindings/swift/TreeSitterScala/scala.h
Comment (eed3si9n) [inline bindings/swift/TreeSitterScala/scala.h:1]: Maybe this file?

Comment (amaanq) [reply bindings/swift/TreeSitterScala/scala.h:1]: It's a swift header file...

Comment (eed3si9n) [reply bindings/swift/TreeSitterScala/scala.h:1]: Yea, currently the sync job is setup to generate and commit bindings file. I think the YAML should be updated to track the expanded sets up binding files:
https://github.com/tree-sitter/tree-sitter-scala/blob/master/.github/workflows/sync.yml

Comment (amaanq) [reply bindings/swift/TreeSitterScala/scala.h:1]: Please don't do that. Bindings require manual intervention to fix missing information that can't be automatically assumed, like the scanner file info and query files.

Comment (eed3si9n) [reply bindings/swift/TreeSitterScala/scala.h:1]: We can relax it for Swing binding file etc, but tree-sitter-scala currently does not manually edit the bindings files as far as I know. Doesn't tree-sitter CLI regularly overwrite these?

Comment (amaanq) [reply bindings/swift/TreeSitterScala/scala.h:1]: Nah, it got moved to the `init` command, because this became problematic and was doing too much

Comment (eed3si9n) [reply bindings/swift/TreeSitterScala/scala.h:1]: ok here's https://github.com/tree-sitter/tree-sitter-scala/pull/432

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**